cmd/devp2p/ethcmd_test.go

```diff
@@ -0,0 +1,156 @@
+)
+
+type testEth struct {
+	*cmdtest.TestCmd
+}
+
+func TestMain(m *testing.M) {
+	// Run the app if we've been exec'd as "ethkey-test" in runEthkey.
+	reexec.Register("devp2p-test", func() {
+		if err := app.Run(os.Args); err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+		os.Exit(0)
+	})
+	// check if we have been reexec'd
+	if reexec.Init() {
+		return
+	}
+	os.Exit(m.Run())
+}
+
+func TestNewBlockAnnouncement(t *testing.T) {
+	geth, err := runGeth()
+	if err != nil {
+		t.Fatalf("could not run geth: %v", err)
+	}
+	defer geth.Close()
+
+	tt := new(testEth)
+	tt.TestCmd = cmdtest.NewTestCmd(t, tt)
+
+	// NewBlock msg with block 1000 from fullchain.rlp.
+	msg := "f90206f901fef901f9a00e70e01064023f70f047dbf0b97e7109e4aa5df3d643d45f8e91e47d3d67a424a01dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347940000000000000000000000000000000000000000a04327919f498aefdfe87f9c5a83cfd4a0bb444973d4a99d44fa9f4728497c3608a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421b9010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000830200008203e884800000008083e51c5880a02807078eecff82eef7ad486076c4989e331a8a4c54592a690f083ef7b24820ec886cad2133e8e98457c0c08407d20000"
+	hash := common.HexToHash("0x8c795a2497f393359fd66bfd5696442d12a81ccb1110dffd636604bfc9af4df3")
+
+	rpc, err := geth.Attach()
+	if err != nil {
+		tt.Fatalf("unable to attach client: %v", err)
+	}
+	ctx := context.Background()
+	client := ethclient.NewClient(rpc)
+
+	// Before the announcement, the propogated block should not be retrievable.
+	_, err = client.BlockByHash(ctx, hash)
+	if err == nil || (err != nil && err.Error() != "not found") {
+		tt.Fatalf("should fail to retrieve block before announcement: %v", err)
+	}
+
+	// Run devp2p eth new-block against node.
+	args := []string{"--verbosity=5", "eth", fmt.Sprintf("--genesis=%s", genesisPath), fmt.Sprintf("--chain=%s", halfchainFile), "new-block", geth.Server().Self().String(), msg}
+	tt.Run("devp2p-test", args...)
+	tt.WaitExit()
+
+	// Give geth a moment to ingest the new block.
+	time.Sleep(time.Second)
+
+	// After the announcement, the block should be readily available.
+	got, err := client.BlockByHash(ctx, hash)
+	if err != nil {
+		tt.Fatalf("unable to get announced block: %v", err)
+	}
+	if hash != got.Hash() {
+		tt.Fatalf("block mismatch (got: %s, want: %s)", hash, got.Hash())
+	}
+}
+
+// runGeth creates and starts a geth node
+func runGeth() (*node.Node, error) {
```

Comment:
It's unfortunate this is mostly copied exactly from `cmd/devp2p/internal/ethtest/suite_test.go`. Not sure if there is anything better that could be done.